C++/libs/Sink/test/src/ThreadSafeSink.cpp

```diff
@@ -0,0 +1,49 @@
+#include <Toolbox/catch.hpp>
+
+#include "TestSink.h"
+#include <Toolbox/Sink/ThreadSafeProxySink.h>
+#include <thread>
+#include <sstream>
+
+using namespace Toolbox::Sink;
+
+TEST_CASE ("ThreadSafeProxySink with lvalue Sink")
+{
+  std::string rc;
+  Test::Sink sink (rc);
+  ThreadSafeProxySink<Test::Sink> proxySink (std::move (sink));
+  std::string message = "test";
+  proxySink << message;
+  
+  REQUIRE (rc == message);
+}
+
+TEST_CASE ("ThreadSafeProxySink with rvalue Sink")
+{
+  std::string rc;
+  ThreadSafeProxySink<Test::Sink> proxySink (Test::Sink {rc});
+  std::string message = "test";
+  proxySink << message;
+  
+  REQUIRE (rc == message);
+}
+
+TEST_CASE ("Instantiate ThreadSafeProxySink using make_thread_safe")
+{
+  std::string rc;
+  ThreadSafeProxySink<Test::Sink> proxySink = make_thread_safe<Test::Sink>(rc);
```

Comment:
I like this.

Reply:
when I write code using ThreadSafeSink, I will opt for auto instead:
```
auto sink = make_thread_safe_sink<Test::Sink>(rc);
```

Reply:
I agree.

Reply:
Also, I have named the function `make_thread_safe` instead of `make_thread_safe_sink` following the same way the smart pointer make functions work, `make_unique` not `make_unique_ptr`